spec/lib/data_update_scripts/migrate_relevant_fields_from_users_to_users_settings_spec.rb

```diff
@@ -0,0 +1,123 @@
+      create_list(:user, 3)
+
+      expect do
+        described_class.new.run
+      end.to change(Users::Setting, :count).from(0).to(3)
+    end
+
+    it "sets the correct User records" do
+      create(:user, config_font: "sans_serif")
+
+      described_class.new.run
+
+      expect(users_setting.config_font).to eq("sans_serif")
+    end
+
+    it "sets the correct Profile records" do
+      profile = create(
+        :profile,
+        :with_DEV_info,
+        user: create(:user, :without_profile),
+        display_email_on_profile: true,
+      )
+
+      described_class.new.run
+
+      expect(users_setting.display_email_on_profile).to eq(profile.display_email_on_profile)
+    end
+
+    it "sets the correct types" do
+      create(:user)
+
+      described_class.new.run
+
+      expect(users_setting.permit_adjacent_sponsors).to be_in([true, false])
+    end
+
+    it "casts data (display_email_on_profile) to the correct types (boolean)" do
+      create(
+        :profile,
+        :with_DEV_info,
+        user: create(:user, :without_profile),
+        display_email_on_profile: true,
+      )
+
+      described_class.new.run
+
+      expect(users_setting.display_email_on_profile).to be_in([true, false])
+    end
+
+    it "sets a fallback value for values that are null" do
+      create(:user, display_announcements: nil)
+
+      described_class.new.run
+
+      expect(users_setting.display_announcements).to be(true)
+    end
+  end
+
+  context "when migrating enum settings" do
+    it "assigns the correct mapping for enum settings" do
+      create(:user, config_font: "monospace")
+
+      described_class.new.run
+
+      expect(users_setting.monospace?).to be(true)
+    end
+
+    it "assigns the fallback value when there value passed does not have an enum defined" do
+      user = create(:user)
+      user.update_columns(config_font: "fake_font")
+      user.reload
+
+      described_class.new.run
+
+      expect(users_setting.default?).to be(true)
+    end
+  end
+
+  it "the updated_at and created_at timestamps are more current than the original values" do
+    user = create(:user, created_at: 1.minute.ago, updated_at: 1.minute.ago)
+
+    described_class.new.run
+
+    expect(users_setting.created_at).to be > user.created_at
+    expect(users_setting.updated_at).to be > user.updated_at
+  end
+
+  context "when the user id exists in both the users_settings users tables" do
+    it "replaces the users_setting values with values from the user table" do
+      user = create(:user, display_announcements: true)
+      user_id = user.id
+
+      described_class.new.run
+
+      expect(users_setting.user_id).to eq(user_id)
+      expect(users_setting.display_announcements).to be(true)
+
+      user.update_columns(display_announcements: false)
+      user.reload
+
+      described_class.new.run
+
+      expect(users_setting.user_id).to eq(user_id)
+      expect(users_setting.display_announcements).to be(false)
```

Comment:
`user.reload` reloads the `User` object, but you're interested in the `User::Settings` value, you have two options:

1. add `user_setting.reload` after `described_class.new.run` so it reloads the value from the DB, which now shall be `false`
2. change the expectation to `expect(user.setting.display_announcements).to be(false)`, this is because `.setting` loads the correct "setting" object from the DB. `User` and `Users::Setting` are related to each other through `user.setting`

Let me know if any of this isn't clear :)